### DISCUSSION TITLE
[Feature Request] Variables: Conditional Macro

### DISCUSSION THREAD

**Discussion by mcarbonneaux** [grafana/grafana#37534] [General]
Can you add conditional macro based on value of variable ?

Exemple of macro `$ifVar` like that with sql request :
```
select * 
from mytable
where 
$ifVar($myvar1, "AND col1 like '%$myvar1%"")
and col2 in ($myvar2)
```

Where `AND col1 like '%$myvar1%"` is added to the request only if `$myvar1` is filled.

issue related: https://github.com/grafana/grafana/issues/14769

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
